### ISSUE TITLE
CMU-MOSEI Word Level Alignement

### ISSUE THREAD

**Issue by XinyuLyu** [A2Zadeh/CMU-MultimodalSDK#59]
Hello,
I think there is somthing wrong in CMU-MultimodalSDK for CMU-MOSEI. Using the SDK, I can only download these two files. And when doing the Word Level Alignement, there is a error. Could you help look into it?
![image](https://user-images.githubusercontent.com/33551398/46229871-0e930800-c335-11e8-8ffd-33646438b958.png)

![image](https://user-images.githubusercontent.com/33551398/46229765-b22fe880-c334-11e8-9d13-2a51dbc4dde9.png)

![image](https://user-images.githubusercontent.com/33551398/46229809-d7bcf200-c334-11e8-95be-49d2f77e1891.png)

**Comment by A2Zadeh** [comment]
@XinyuLyu @Desperado000 yeah I will solve this issue. The problem is that one of the files seemed to have crashed for feature extraction, so I have to make sure and see what happened there. Namely video id 4kF0U8kIMp4 has no features extracted in covarep and facet. I will also add a functionality to unify the keys of the computational sequences, so at least you won't get this error (even if a feature extraction has failed)

**Comment by XinyuLyu** [comment]
Thank you so much!

**Comment by A2Zadeh** [comment]
This issue should now be solved. Before any align, simply call the intersect function on your dataset. For example:

`mydataset.intersect()`

This removes any ids that are not common.

**Comment by XinyuLyu** [comment]
Hello, 
I see your idea, but I don't know how t code it exactly. Could you add specific codes to my word-level-alignement funtion?
<code>def word_level_alignement():</code>
    <code>cmumosei_highlevel = mmdatasdk.mmdataset(mmdatasdk.cmu_mosei.highlevel, 'cmumosei/')</code>
   <code> cmumosei_highlevel.align('glove_vectors')</code>
    <code>cmumosei_highlevel.add_computational_sequences(mmdatasdk.cmu_mosei.labels, 'cmumosei/')</code>
    <code>cmumosei_highlevel.align('Opinion Segment Labels')</code>

**Comment by XinyuLyu** [comment]
Hello,
You mean you have updated the codes in SDK, or we should add the intersect codes by ourselves.

**Comment by XinyuLyu** [comment]
Hello,
I pull the SDK again running the codes about 6 hrs, but it failed again with the same error with different dataset. Please fix it. Many thanks. And why the SDK is running so slow now?
![image](https://user-images.githubusercontent.com/33551398/46447413-ac307200-c74f-11e8-83f6-218f5b0d97eb.png)

**Comment by A2Zadeh** [comment]
Hmm. Let me look into this. To begin with the ID 4kF0U8kIMp4 should have been removed.

**Comment by A2Zadeh** [comment]
@XinyuLyu did you run `git pull`?

**Comment by XinyuLyu** [comment]
I have pulled it, let me try again.

**Comment by A2Zadeh** [comment]
@XinyuLyu awesome, keep me in the loop.

**Comment by A2Zadeh** [comment]
@XinyuLyu did the issue get resolved?

**Comment by XinyuLyu** [comment]
No, it failed again. I delete the previous SDK project and git clone it again. Then I run it...
![image](https://user-images.githubusercontent.com/33551398/46557524-2b40ba00-c8b8-11e8-8ea3-49736533787b.png)

**Comment by A2Zadeh** [comment]
@XinyuLyu very strange. I just had the data aligned again. On which data point did it hit the problem? Do you see a yellow line before alignment starts saying a data point has been removed?

**Comment by XinyuLyu** [comment]
Word aligment has finished, but there is an another error with the code <code> >>>cmumosi_highlevel.add_computational_sequence(mmdatasdk.cmu_mosi.labels,'cmumosi/')</code>.
![image](https://user-images.githubusercontent.com/33551398/46566597-85a93d00-c8ef-11e8-8635-736548774b8a.png)

**Comment by Desperado000** [comment]
Switch python 2 or modify the dataset.py @XinyuLyu

**Comment by A2Zadeh** [comment]
@Desperado000 oh you think this is related to python2 and 3?

**Comment by Desperado000** [comment]
@A2Zadeh iteritems() has been removed in python 3. 
It takes so long to align the data. Is there any way to storge the aligned data in binary format? Thanks.

**Comment by A2Zadeh** [comment]
@Desperado000 Ah yeah I should fix that. Simply call deploy on the dataset object and never worry about aligning again. You can store the aligned files and load them again in a dataset object. I am going to add the description for it in the examples that I will upload.

**Comment by A2Zadeh** [comment]
@XinyuLyu I think you have a very old version of the sdk. iteritems changed to items at least few weeks ago. Can you pull the SDK again?

**Comment by A2Zadeh** [comment]
> Word aligment has finished, but there is an another error with the code ` >>>cmumosi_highlevel.add_computational_sequence(mmdatasdk.cmu_mosi.labels,'cmumosi/')`.
> ![image](https://user-images.githubusercontent.com/33551398/46566597-85a93d00-c8ef-11e8-8635-736548774b8a.png)

This bug should be fixed. Please let me know if you encounter the same problem again.

**Comment by XinyuLyu** [comment]
Thank you so much! And the last question about the CMU-MOSEI Word Level Alignement. There is a bug in the last step, with the code <code>cmumosei_highlevel.align('Opinion Segment Labels')</code>. Could you help me with that? 
![image](https://user-images.githubusercontent.com/33551398/46637384-fd62ac00-cb29-11e8-95cd-e199617d50d0.png)

**Comment by A2Zadeh** [comment]
@XinyuLyu You are welcome!

1. did you download the labels? `cmumosi_highlevel.add_computational_sequence(mmdatasdk.cmu_mosi.labels,'cmumosi/')`

2. most likely the file is not there. It could be in a folder. Like 'mydataset/CMU_MOSEI_VisualFacet42.csd'. I should clarify that error better.

**Comment by XinyuLyu** [comment]
Hello,
I don't download the labels of 'CMU_MOSEI' by myself from the website in cmu_mosei.py, for I think the code  will download the label file for me with the code in next line.
<code>>>>cmumosei_highlevel=mmdatasdk.mmdataset(mmdatasdk.cmu_mosei.highlevel,'cmumosei/')</code>. 
And when I finished the codes below, in 'cmumosei/' I have such five csd files. I am not sure if which one is the label file.
![image](https://user-images.githubusercontent.com/33551398/46694780-b7662080-cbdb-11e8-8a34-dd6f9a7ca0e8.png)

 
![image](https://user-images.githubusercontent.com/33551398/46694590-4c1c4e80-cbdb-11e8-85f0-e12f36f9efa7.png)

**Comment by A2Zadeh** [comment]
@XinyuLyu The two files starting with labels are the ones you are looking for.

**Comment by XinyuLyu** [comment]
Thank you! Since I have downloaded the labels, why the error happens, in the last step of word aligment?
![image](https://user-images.githubusercontent.com/33551398/47240817-7f1ec900-d3b7-11e8-9930-3b4b19e828bb.png)

**Comment by A2Zadeh** [comment]
Yes, there is no Opinion Segment Labels in the MOSEI. When you load the labels you can see what the dataset ends up referring to the labels. "Sentiment Labels" and "Emotion Labels" is what you are looking for.
